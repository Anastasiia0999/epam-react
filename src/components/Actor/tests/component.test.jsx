import React from 'react';
import { mount, shallow } from 'enzyme';

import {Actor} from "../index";

describe('<Actor />', () =>{
    let wrapper;
    let instance;

    const props = {
        fields:{
            "actor-name": "Actor",
            "bio": "Biography"
        },
        openedActor:{
            "id": 37,
            "name": "Joseph Gordon-Levitt",
            "imgUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTQzOTg0NTkzNF5BMl5BanBnXkFtZTcwNTQ4MTcwOQ@@._V1_UY317_CR35,0,214,317_AL_.jpg",
            "biography": "Lorem ipsum dolor sit amet"
        },
        loadingFlag: false,
        params: null
    };

    beforeEach(() => {
        wrapper = mount(<Actor {...props} />);
        instance = wrapper.instance();
    });

    it('should render correctly', () =>{
        expect(wrapper.html()).toMatchSnapshot();
    });


    it('correct author`s name render', () =>{
        expect(props.openedActor.name).toBe("Joseph Gordon-Levitt");
    });

    it('correct author`s photo link render', () =>{
        expect(props.openedActor.imgUrl).toBe("https://images-na.ssl-images-amazon.com/images/M/MV5BMTQzOTg0NTkzNF5BMl5BanBnXkFtZTcwNTQ4MTcwOQ@@._V1_UY317_CR35,0,214,317_AL_.jpg");
    });

    it('correct author`s biography render', () =>{
        expect(props.openedActor.biography).toBe("Lorem ipsum dolor sit amet");
    });

});


