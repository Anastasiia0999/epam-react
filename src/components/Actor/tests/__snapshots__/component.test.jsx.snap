// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actor /> should render correctly 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"card card-box col-8\\"><div class=\\"card-header movie-box_header my-3\\"><h2>Actor: Joseph Gordon-Levitt</h2></div><div class=\\"card-body actor-card\\"><div class=\\"col-6\\"><img src=\\"https://images-na.ssl-images-amazon.com/images/M/MV5BMTQzOTg0NTkzNF5BMl5BanBnXkFtZTcwNTQ4MTcwOQ@@._V1_UY317_CR35,0,214,317_AL_.jpg\\" class=\\"rounded actor-box_image\\" alt=\\"poster\\"></div><div class=\\"col-6\\"><p class=\\"biography\\" data-spy=\\"scroll\\"><b>Biography: </b> Lorem ipsum dolor sit amet</p></div></div></div></div></div>"`;
